Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputSystem.cs

```diff
@@ -244,24 +244,55 @@ public override void Disable()
         /// <inheritdoc />
         public override void HandleEvent<T>(BaseEventData eventData, ExecuteEvents.EventFunction<T> eventHandler)
         {
+            Debug.Assert(eventData != null);
+            Debug.Assert(!(eventData is MixedRealityPointerEventData), "HandleEvent called with a pointer event. All events raised by pointer should call HandlePointerEvent");
+
             if (disabledRefCount > 0)
             {
                 return;
             }
 
-            Debug.Assert(eventData != null);
             var baseInputEventData = ExecuteEvents.ValidateEventData<BaseInputEventData>(eventData);
-            Debug.Assert(baseInputEventData != null);
-            Debug.Assert(!baseInputEventData.used);
+            DispatchEventToGlobalListeners(baseInputEventData, eventHandler);
 
-            if (baseInputEventData.InputSource == null)
+            if (baseInputEventData.used)
             {
-                Debug.LogError($"Failed to find an input source for {baseInputEventData}");
+                // All global listeners get a chance to see the event,
+                // but if any of them marked it used,
+                // we stop the event from going any further.
                 return;
             }
 
-            // Send the event to global listeners
-            base.HandleEvent(eventData, eventHandler);
+            Debug.Assert(baseInputEventData.InputSource.Pointers != null, $"InputSource {baseInputEventData.InputSource.SourceName} doesn't have any registered pointers! Input Sources without pointers should use the GazeProvider's pointer as a default fallback.");
+
+            var modalEventHandled = false;
+            // Get the focused object for each pointer of the event source
+            for (int i = 0; i < baseInputEventData.InputSource.Pointers.Length && !baseInputEventData.used; i++)
+            {
+                modalEventHandled = DispatchEventToObjectFocusedByPointer(baseInputEventData.InputSource.Pointers[i], baseInputEventData, modalEventHandled, eventHandler);
```

Comment:
is this used anywhere after being assigned?

Reply:
Yes, on that line it's passed into the `DispatchEventToObjectFocusedByPointer` method.

Reply:
Oh I see, it's a loop. Thanks :)